java/flight/src/main/protobuf/flight.proto

```diff
@@ -0,0 +1,111 @@
+syntax = "proto3";
+
+option java_package = "org.apache.arrow.flight.impl";
+
+service FlightService {
+
+  // get a list of available streams.
+  rpc ListFlights(Criteria) returns (stream FlightGetInfo) {}
+  
+    // get a list of available streams.
+  rpc GetFlightInfo(FlightDescriptor) returns (FlightGetInfo) {}
+  
+  // get stream data.
+  rpc DoGet(Ticket) returns (stream FlightData) {}
+
+  // create or append a stream. 
+  rpc DoPut(stream FlightData) returns (PutResult) {}
+  
+  rpc DoAction(Action) returns (Result) {}
+  
+  rpc ListActions(Empty) returns (stream ActionType) {}
+  
+}
+
+message Empty {}
+
+message ActionType {
+  string type = 1;
+  string description = 2;
+}
+
+message Criteria {
+  // TODO: criteria pattern to be added.
+  int64 slice_target = 1;  
+}
+
+message Action {
+  string type = 1;
+  bytes body = 2;
+}
+
+message Result {
+  string type = 1;
+  bytes body = 2;
+}
+
+message FlightDescriptor {
```

Comment:
It is not very clear to me what is the meaning of this class. Would you mind elaborating a bit?